### PR TITLE
[CAST-154] Kubesec OWASP Unrestricted Resource Consumption

### PR DESCRIPTION
**Pull request by justincorsha**
Before you submit your PR, make sure you check each of the following:

1. [x] I have read the [Contributing Guidelines](./CONTRIBUTING.md) and signed the [Contributor License Agreement](./CLA.md).
2. [x] My code is clean and ready-to-PR, including being free of lint errors and dead code
3. [x] I have **tested** my change / code and am confident that it works
4. [x] I have filled in the template below completely and accurately
##### Changelog

- Duplicated KubesecFinding into KubesecResourcesFinding
- Created kubesec-resources analysis to look for Rule IDs relating to resource requests/limits
- Removed resource related findings from combined Kubesec table

##### Testing

- Deployed and checked that Rules with IDs `RequestsMemory`, `RequestsCPU`, `LimitsMemory`, `LimitsCPU` or `VolumeClaimRequestsStorage` appear in the new "Kubesec: Unrestricted Resource Consumption" table instead of the "Kubesec" table
 
Screenshot 2023-09-05 at 4 40 57 PM

##### Unit Tests

Added snapshot test for KubesecCard

### CODE REVIEW

**Review by JasonCorsha — APPROVED** [review]
LGTM

File: ui/lib/analysis/kubesec-resources.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+   const findingsQuery = `
+   SELECT
+     id as id,
+     REPLACE(plugin_name, 'cast-kubesec', 'cast-kubesec-resources') as type,
+     occurred_at as detectedAt,
+     data
+   FROM plugins_findings
+   WHERE 
+     plugin_name = 'cast-kubesec' and 
+     data->>'Rule ID' IN ('${kubesecResourcesRules.join('\', \'')}')
```
Comment (ArcticLight) [inline ui/lib/analysis/kubesec-resources.ts:34]: This needs to be a parameterized query using a query builder, any time you're interpolating raw text.
I.e.:

```ts
// ...
findingsQuery = `
 data->>'Rule ID' IN $1
`;

// ...
const { rows } = await conn.query(query, [kubesecResourcesRules]);
```

It's much safer this way and you don't have to worry about implementing quoting correctly.
See: https://node-postgres.com/features/queries

File: ui/lib/analysis/kubesec-types.ts [hunk 2]
```diff
@@ -55,3 +55,46 @@ export const kubesecRowToFinding = ({
...
+  data,
+});
+
+export const kubesecResourcesRules: string[] = [
+  'RequestsMemory', 
+  'RequestsCPU',
+  'LimitsMemory',
+  'LimitsCPU',
+  'VolumeClaimRequestsStorage', 
+ ];
```
Comment (ArcticLight) [inline ui/lib/analysis/kubesec-types.ts:100]: This should probably be moved from the types file (which is primarily concerned with Zod schemas) and into the kubesec analysis module instead, where this is used.

File: ui/lib/analysis/kubesec.ts [hunk 3]
```diff
@@ -29,7 +29,7 @@ SELECT
   occurred_at as detectedAt,
   data
 FROM plugins_findings
-WHERE plugin_name = 'cast-kubesec'
+WHERE plugin_name = 'cast-kubesec' and not data->>'Rule ID' IN ('${kubesecResourcesRules.join('\', \'')}')
```
Comment (ArcticLight) [inline ui/lib/analysis/kubesec.ts:32]: Same comment about using Query variables as above

**Review by ArcticLight — CHANGES_REQUESTED** [review]

File: ui/lib/analysis/kubesec-resources.ts (hunk 1 above)
Comment (justincorsha) [reply ui/lib/analysis/kubesec-resources.ts:34]: Great catch! This is a much cleaner solution too.

**Review by ArcticLight — APPROVED** [review]

**Pull request merged**